pkg/controller/jobs/job/job_controller.go

```diff
@@ -153,9 +157,26 @@ func (j *Job) GetGVK() schema.GroupVersionKind {
 	return gvk
 }
 
+func (j *Job) ReclaimablePods() []kueue.ReclaimablePod {
+	if j.Spec.Parallelism == nil || *j.Spec.Parallelism == 1 || j.Spec.Completions == nil || j.Status.Succeeded == 0 {
+		return nil
+	}
+
+	remaining := *j.Spec.Completions - j.Status.Succeeded
+	if remaining >= *j.Spec.Parallelism {
+		return nil
+	}
+
+	return []kueue.ReclaimablePod{{
+		Name:  podSetName,
+		Count: *j.Spec.Parallelism - remaining,
+	}}
+}
+
 func (j *Job) PodSets() []kueue.PodSet {
 	return []kueue.PodSet{
 		{
+			Name:     podSetName,
```

Comment:
```suggestion
```

We set Workload name in defaulting webhook.

https://github.com/kubernetes-sigs/kueue/blob/f5bb4a263e92f92c5d483bcd4171778fac7b4510/apis/kueue/webhooks/workload_webhook.go#L57C56-L64

Reply:
Yes, but before the name was not relevant for this controller. If we don't specifically set it here and for some reason in the future the default name is changed, it will break the reporting of reclaimable pods .

Reply:
SGTM